src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnectionBuilderExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.SignalR.Client
+{
+    public static class HubConnectionBuilderExtensions
+    {
+        public static IHubConnectionBuilder WithHubProtocol(this IHubConnectionBuilder hubConnectionBuilder, IHubProtocol hubProtocol)
```

Comment:
Should we add WithJsonProtocol and WithMessagePackProtocol?

Reply:
I actually had the same thought yesterday - added.